geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/AbstractFileSystemStorage.scala

```diff
@@ -235,8 +235,8 @@ abstract class AbstractFileSystemStorage(
       val path = StorageUtils.nextFile(context.root, partition, metadata.leafStorage, extension, fileType)
       PathCache.register(context.fc, path)
       val updateObserver = new UpdateObserver(partition, path, action)
-      val observer = if (observers.isEmpty) { updateObserver } else {
-        new CompositeObserver(observers.map(_.apply(path)).+:(updateObserver))
+      val observer = if (observers.isEmpty) { updateObserver.asInstanceOf[BoundsObserver] } else {
+        new CompositeObserver(observers.map(_.apply(path)).+:(updateObserver)).asInstanceOf[BoundsObserver]
```

Comment:
i think ideally we could use the bounds that the writer is already calculating, so that we don't have to calculate them twice.